### PR TITLE
migrate s3 upload url to skill-api and skill-lesson

### PR DESCRIPTION
**Pull request by joelhooks**
uploader

### CODE REVIEW

File: packages/skill-api/src/core/index.ts [hunk 1]
```diff
@@ -72,6 +73,8 @@ export async function SkillRecordingsHandler<
         return render.test()
       case 'subscriber':
         return await convertkitLoadSubscriber({params})
+      case 'sign-s3':
```
Comment (joelhooks) [inline packages/skill-api/src/core/index.ts:76]: `/api/skill/sign-s3`

File: apps/total-typescript/src/module-builder/video-uploader.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import * as React from 'react'
-import {useFileChange} from './use-file-change'
-import {uploadToS3} from './upload-file'
+import {uploadToS3} from '@skillrecordings/skill-lesson/lib/upload-file'
+import {useFileChange} from '@skillrecordings/skill-lesson/lib/use-file-change'
 
 const VideoUploader = () => {
```
Comment (joelhooks) [inline apps/total-typescript/src/module-builder/video-uploader.tsx:5]: @vojtaholik goal is to make this a core ui component, but I'd like to have it handle multiple files at once and do a queue.

File: apps/epic-web/src/pages/api/skill/[...skillRecordings].ts [hunk 3]
```diff
@@ -1,14 +1,22 @@
...
+import {NextApiRequest} from 'next'
+import {getCurrentAbility, UserSchema} from '@skillrecordings/skill-lesson'
 
 export const skillOptions: SkillRecordingsOptions = {
   site: {
     title: process.env.NEXT_PUBLIC_SITE_TITLE,
     supportEmail: process.env.NEXT_PUBLIC_SUPPORT_EMAIL,
   },
   nextAuthOptions,
+  getAbility: async (req: IncomingRequest) => {
```
Comment (joelhooks) [inline apps/epic-web/src/pages/api/skill/[...skillRecordings].ts:16]: this let's use use abilities in a contextual way inside of skill-api for stuff that needs authorization

File: packages/skill-api/src/core/lib/get-signed-url.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  region: process.env.AWS_VIDEO_UPLOAD_REGION,
+  credentials: {
+    accessKeyId: process.env.AWS_VIDEO_UPLOAD_ACCESS_KEY_ID!,
+    secretAccessKey: process.env.AWS_VIDEO_UPLOAD_SECRET_ACCESS_KEY!,
+  },
+}
+
+const client = new S3Client(options)
+
+export async function getSignedUrlForVideoFile(options: {filename: string}) {
```
Comment (joelhooks) [inline packages/skill-api/src/core/lib/get-signed-url.ts:15]: uses the aws-sdk v3 to get a signed url for a specific filename

File: packages/skill-api/src/core/services/aws.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+export async function signs3UploadUrl({
+  params,
+  token,
+}: {
+  params: SkillRecordingsHandlerParams
+  token: JWT | null
+}): Promise<OutgoingResponse> {
+  if (params.options.getAbility) {
+    const ability = await params.options.getAbility(params.req)
```
Comment (joelhooks) [inline packages/skill-api/src/core/services/aws.ts:14]: I didn't make getAbility generic, but that'd be rad

**Review by vojtaholik — APPROVED** [review]

**Pull request merged**